### ISSUE TITLE
Compatibility with pulseaudio-ctl

### ISSUE THREAD

**Issue by somini** [smasher164/pw-volume#6]
https://github.com/graysky2/pulseaudio-ctl/

This is a similar program that exists for pulseaudio control. Since these tools live and die by the interface, I would request this interface to be implemented.

```
/usr/bin/pulseaudio-ctl mute        ==>  Toggle status of mute
/usr/bin/pulseaudio-ctl mute-input  ==>  Toggle status of mute for mic
/usr/bin/pulseaudio-ctl up          ==>  Increase vol by 5 %
/usr/bin/pulseaudio-ctl up 25       ==>  Increase vol by 25 %
/usr/bin/pulseaudio-ctl down        ==>  Decrease vol by 5 %
/usr/bin/pulseaudio-ctl down 50     ==>  Decrease vol by 50 %
/usr/bin/pulseaudio-ctl set 40      ==>  Set vol to 40%
/usr/bin/pulseaudio-ctl atmost 30   ==>  Set vol to 30% if current higher than that
```

From what I can tell, these are mostly aliases, there are only one crucial feature to implement:
- `mute` -> `mute toggle`
- `mute-input`: To Implement
- `up`/`up X` -> `change +X`
- `down`/`down X` -> `change -X`
- `set X` -> `change X` (not sure this is supported`
- `atmost X`: To Implement. Can be hacked around by reading the status, then writing the new value, it's lower priority.

**Comment by somini** [comment]
There's also this issue to implement the opposite of this, pipewire support on `pulseaudio-ctl`:

https://github.com/graysky2/pulseaudio-ctl/issues/68

**Comment by smasher164** [comment]
Thanks for filing this. I didn't think about maintaining compatibility with pulseaudio-ctl.

Input volume control is something that should definitely be added. The default input sink is something that can be retrieved via pw-dump, so it should be the same sort of logic.

Atmost is interesting. Is that something people actually use?

**Comment by somini** [comment]
> Atmost is interesting. Is that something people actually use?

I don't use it personally, but it might be very useful for automation. For example, limiting the audio level when connecting new devices, or when on calls?

**Comment by somini** [comment]
FYI, looks like WirePlumber "solved" this issue by shipping a program that does basically the same thing:

https://wiki.archlinux.org/title/WirePlumber#Keyboard_volume_control

**Comment by smasher164** [comment]
Maybe I should redirect users to use `wpctl` instead?

**Comment by somini** [comment]
Yeah, I think WirePlumber and PipeWire are almost sister projects, it's the default session manager.

**Comment by eternal-sorrow** [comment]
Not everyone uses wireplumber with pipewire, please don't abandon this project.
